pkg/tls/tls.go

```diff
@@ -72,12 +80,23 @@ func getCertificates(clientCertFile, clientKeyFile string) ([]tls.Certificate, e
 	return []tls.Certificate{}, nil
 }
 
-func readFile(path string) ([]byte, error) {
-	file, err := os.Open(path)
-	if err != nil {
-		return nil, err
+// readPEMBytes will parse the certificate on input and return the pem byte
+// content. It accepts a pem certificate or the file path to a pem certificate.
+func readPEMBytes(cert string) ([]byte, error) {
+	if isPEMString(cert) {
+		return []byte(cert), nil
 	}
-	defer file.Close()
 
-	return io.ReadAll(file)
+	return os.ReadFile(cert)
+}
+
+// isPEMString returns true if the provided string match a PEM formatted certificate.
+func isPEMString(s string) bool {
+	// trim the certificates to make sure we tolerate any yaml weirdness
+	trimmedStr := strings.TrimSpace(s)
+
+	// we assume that the string starts with "-" and let further validation
+	// verifies the PEM format. When migrating from pkcs12 to PEM a "Bag
```

Comment:
```suggestion
	// verify the PEM format. When migrating from pkcs12 to PEM a "Bag
```

Reply:
I agree, it's a bit confusing. I added this as it's a supported functionality in the xata codebase, so it would otherwise have trouble connecting with the given config. I can update it so that it's different methods to be more explicit about what's being passed (originally this library only supported files for example).